### PR TITLE
Update script for yamllint

### PR DESCRIPTION
**Pull request by averikitsch**
yamllint must be run in the directory of the file. This is the only tool I can find, especially with a provided docker image, but we should consider removing or finding a better tool due to a lack of flexibility. There also are issues around using a config file not in the directory as well.

### CODE REVIEW

**Comment by grayside** [comment]
> There also are issues around using a config file not in the directory as well.

```
yamllint -c path/to/config?
```

**Comment by averikitsch** [comment]
> > There also are issues around using a config file not in the directory as well.
> 
> Would this help?
> 
> ```
> yamllint -c path/to/config
> ```

In practice this feature seemed to be broken

File: yaml.lint.cloudbuild.yaml [hunk 1]
```diff
@@ -4,9 +4,17 @@ steps:
...
-      | grep -q 'error'
-    if [[ $? -eq 0 ]] ; then exit 1 ; fi
+  - | # Recursively find yaml files (yamllint must be run in directory)
+    find . -name "*.yaml" -not -path "./testing/*" -not -path "*.lint.cloudbuild.yaml" \
+      -exec sh -c '
+      root=$(pwd)
+      lint () {
+        cd "$root" || exit
+        dir=$(dirname "$@")
+        cd "$dir" || exit
```
Comment (grayside) [inline yaml.lint.cloudbuild.yaml:14]: pushd and popd are good alternatives to this, as you don't need to preserve PWD state, just identify the directory you need to visit. Example: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/testing/kokoro/system_tests.sh#L72

File: yaml.lint.cloudbuild.yaml [hunk 2]
```diff
@@ -4,9 +4,17 @@ steps:
...
-    find ./* -name '*.yaml' -not -path "./testing/*" \
-      -exec yamllint -f parsable {} \; \
-      | tee /proc/1/fd/1 \
-      | grep -q 'error'
-    if [[ $? -eq 0 ]] ; then exit 1 ; fi
+  - | # Recursively find yaml files (yamllint must be run in directory)
+    find . -name "*.yaml" -not -path "./testing/*" -not -path "*.lint.cloudbuild.yaml" \
+      -exec sh -c '
+      root=$(pwd)
+      lint () {
```
Comment (grayside) [inline yaml.lint.cloudbuild.yaml:11]: I think defining the function to be called before the find would make the whole thing much more readable.

File: shell.lint.cloudbuild.yaml [hunk 3]
```diff
@@ -5,8 +5,8 @@ steps:
   args:
   - -c
   - | # Recursively find shell scripts (not in testing dir)
-    find . -name '*.sh' -not -path './testing/*' \
+    find . -name "*.sh" -not -path "./testing/*" \
```
Comment (grayside) [inline shell.lint.cloudbuild.yaml:8]: globs strike again?

File: yaml.lint.cloudbuild.yaml (hunk 2 above)
Comment (averikitsch) [reply yaml.lint.cloudbuild.yaml:11]: It seems like there are limitations around moving the function outside the find when using shell (https://unix.stackexchange.com/questions/50692/executing-user-defined-function-in-a-find-exec-call)

File: shell.lint.cloudbuild.yaml (hunk 3 above)
Comment (averikitsch) [reply shell.lint.cloudbuild.yaml:8]: This seems safer and more consistent. Can we chat about this in our next 1:1?

Comment (averikitsch) [reply shell.lint.cloudbuild.yaml:8]: In practice it doesn't seem to have an effect, but for overall consistency should I remove the quotes?

File: yaml.lint.cloudbuild.yaml [hunk 4]
```diff
@@ -4,9 +4,17 @@ steps:
...
+        cd "$root" || exit
```
Comment (averikitsch) [inline yaml.lint.cloudbuild.yaml:12]: pushd and popd won't work in the shell environment

Comment (grayside) [reply yaml.lint.cloudbuild.yaml:12]: That's surprising. Maybe it would require `bash`? Fine to keep `cd`.

Comment (averikitsch) [reply yaml.lint.cloudbuild.yaml:12]: It does require bash but I don't believe that comes installed in the alpine image.

**Comment by grayside** [comment]
Merge when ready.

**Pull request merged**